### ISSUE TITLE
[v3]  Implement new Concept Exercise: modules

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#851]
This issue describes how to implement the `modules` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Modules in F#.

## Learning objectives

- Know what modules are.
- Know how to define a module.
- Know about nested modules.
- Know what namespaces are.
- Know how to import namespaces.
- Know how to use access control to limit access to elements.

## Out of scope

- Assemblies.
- Know about recursive modules.

## Concepts

The Concepts this exercise unlocks are:

- `modules`: know what modules are; know how to define a module; know about nested modules; know about recursive modules.
- `namespaces`: know what namespaces are; know how to import namespaces.
- `access-control`: know how to use access control to limit access to elements.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

## Prerequisites

This exercise's prerequisites Concepts are:

- `basics`: know how to define bindings, know how to use `int` values.

### Hints

- [Modules][modules]: what modules are and how to define them.
- [Nested modules][nested-modules]: describes nested modules.
- [Recursive modules][recursive-modules]: describes recursive modules.
- [Namespaces][namespaces]: what namespaces are and how to define them.
- [Imports][imports]: how to import modules/namespaces.
- [Access control][access-control]: describes how access control works.

### After

- [Modules][modules]: what modules are and how to define them.
- [Nested modules][nested-modules]: describes nested modules.
- [Recursive modules][recursive-modules]: describes recursive modules.
- [Namespaces][namespaces]: what namespaces are and how to define them.
- [Imports][imports]: how to import modules/namespaces.
- [Access control][access-control]: describes how access control works.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/types/map.md
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[modules]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/modules
[nested-modules]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/modules#nested-modules
[recursive-modules]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/modules#recursive-modules
[namespaces]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/namespaces
[imports]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/import-declarations-the-open-keyword
[access-control]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/access-control
